draft-ietf-quic-transport.md

```diff
@@ -2618,6 +2633,39 @@ advertise a smaller maximum ID.  A sender may receive MAX_STREAM_ID frames out
 of order; a sender MUST therefore ignore any MAX_STREAM_ID that does not
 increase the maximum.
 
+
+## Related Streams
+
+Using the Related Stream ID in the STREAM frame ({{frame-stream}}) allows for
+the creation of a pair of streams.  One endpoint creates a stream without
+assigning a related stream and the peer opens another stream that includes the
+stream ID of the first in the Related Stream ID of the first stream.  This
+enables straightforward use of QUIC for request-response protocols or protocols
+that rely on bidirectional channels.
+
+This document doesn't proscribe the use of the Related Stream ID field, leaving
+decisions about the semantics of related streams to protocols that use QUIC.
+Protocols that use QUIC MUST either define the semantics of related streams or
+prohibit the use of the Related Stream ID field.  Protocols MAY define other
+uses for the Related Stream ID field, but are advised that complex relationships
+between streams could be poorly supported in generic QUIC implementations.  The
+INVALID_RELATED_STREAM error code can be used by protocols to signal receipt of
+invalid Related Stream ID field values, unless the protocol defines a more
+specific error code for the circumstances.
+
+The Related Stream ID for a stream only needs to be sent once for each (allowing
+for retransmissions due to loss).  Sending the field on all stream frames with
+an offset of 0 ensures that the value is received without duplicating the field
+unnecessarily.  If the value of the Related Stream ID changes for a stream, the
+endpoint that receives the conflicting value MUST treat this as a connection
+error of type INVALID_RELATED_STREAM.
```

Comment:
I don't think the updated text explicitly says whether **NOT** setting the `R` bit on subsequent frames constitutes a change.  I think the simplest way to scope it is that if there's a related stream, `R` MUST be set on the zero-offset frame and MAY be set on other frames, but the value of the Related Stream ID MUST NOT change on subsequent frames if `R` is set.

If you receive frames out-of-order and `R` is set, you have the info early; if it's *not* set then you can't do anything with that yet because it might still have been set on some other frame.  You still have to wait for the zero-offset frame to conclude that it will never be set.